File: lab/python/palindromos/palindromo_numerico.py
def inverter_numero(number):
    inverso = 0
    while number > 0:
        resto = number % 10
        inverso = (inverso * 10) + resto
        number //= 10
    return inverso


def is_palindromo(number):
    if not isinstance(number, int):
        return False
    number = abs(number)
    return number == inverter_numero(number)


def main():
    numbers = [121, 242, 123, 321, 645, 111, 9999, 1000, -98, -989, "teste"]
    for number in numbers:
        print(
            number,
            "->",
            "Palíndromo" if is_palindromo(number) else "Não é palíndromo"
        )


main()

File: lab/javascript/palindromos/palindromo_numerico.js
function inverter_numero(number) {
  let inverso = 0;
  while (number > 0) {
    const resto = number % 10;
    inverso = inverso * 10 + resto;
    number = Math.floor(number / 10);
  }
  return inverso;
}

function is_palindromo(number) {
  if (typeof number !== "number" || !Number.isInteger(number)) {
    return false;
  }
  number = Math.abs(number);
  return number === inverter_numero(number);
}

function main() {
  const numbers = [121, 242, 123, 321, 645, 111, 9999, 1000, -98, -989, "teste"];
  for (const number of numbers) {
    console.log(
      `${number} -> ${is_palindromo(number) ? "Palíndromo" : "Não é palíndromo"}`
    );
  }
}

main();
